Source/WebCore/platform/cocoa/NetworkExtensionContentFilter.mm

```diff
@@ -114,6 +117,55 @@
     request.setURL(WTFMove(modifiedRequestURL));
 }
 
+void NetworkExtensionContentFilter::willSendRequest(ResourceRequest&& request, const ResourceResponse& redirectResponse, CompletionHandler<void(ResourceRequest, State)>&& completionHandler)
+{
+    ASSERT(!request.isNull());
+    if (!request.url().protocolIsInHTTPFamily() || !enabled()) {
+        m_state = State::Allowed;
+        completionHandler(WTFMove(request), m_state);
+        return;
+    }
+
+    initialize();
+
+    if (!redirectResponse.isNull()) {
+        responseReceived(redirectResponse);
+        if (!needsMoreData()) {
+            completionHandler(WTFMove(request), m_state);
+            return;
+        }
+    }
+
+    RetainPtr protectedNSURLRequest = request.protectedNSURLRequest(DoNotUpdateHTTPBody);
+
+    auto blockPtr = makeBlockPtr([protectedThis = Ref { *this }, request = crossThreadCopy(WTFMove(request)), completionHandler = WTFMove(completionHandler)](NEFilterSourceStatus status, NSDictionary *decisionInfo) mutable {
+        RetainPtr<NSString> modifiedRequestURLString;
+        if (decisionInfo)
+            modifiedRequestURLString = decisionInfo[NEFilterSourceOptionsRedirectURL];
```

Comment:
We probably want to use `checked_objc_cast<NSString>()` here instead of the assertion below.